pkg/kubelet/config/config.go

```diff
@@ -238,16 +238,7 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 			}
 			recordFirstSeenTime(ref)
 			pods[name] = ref
-			// If a pod is not found in the cache, and it's also not in the
-			// pending phase, it implies that kubelet may have restarted.
-			// Treat this pod as update so that kubelet wouldn't reject the
-			// pod in the admission process.
-			if ref.Status.Phase != api.PodPending {
-				updatePods = append(updatePods, ref)
-			} else {
-				// this is an add
-				addPods = append(addPods, ref)
-			}
+			addPods = append(addPods, ref)
```

Comment:
@dchen1107 FYI, This part of code is introduced in https://github.com/kubernetes/kubernetes/commit/712612c2dc3d54abd50e8d9db35fa5140c843d46#diff-a26895e2583004f1c326cb890c6aac5c